src/VisualStudio/Core/Impl/CodeModel/FileCodeModel.cs

```diff
@@ -189,14 +189,22 @@ internal void UpdateCodeElementNodeKey(AbstractKeyedCodeElement keyedElement, Sy
                 throw new InvalidOperationException($"Unexpected failure in Code Model while updating node keys {oldNodeKey} -> {newNodeKey}");
             }
 
+            // If we're updating this element with the same node key as an element that's already in the table,
+            // just remove the old element. The old element will continue to function (through its node key), but
+            // the new element will replace it in the cache.
+            if (_codeElementTable.ContainsKey(newNodeKey))
+            {
+                _codeElementTable.Remove(newNodeKey);
```

Comment:
so, issue is it used to throw?

http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/Interop/CleanableWeakComHandleTable.cs,180